File: programmers/코딩테스트 고득점 Kit/스택, 큐/주식가격.java
import java.util.*;
class Solution {
    public int[] solution(int[] prices) {
        int[] answer = new int[prices.length];
        int Min = 100000;
        int last = 100000;
        Stack<Integer> seconds = new Stack<>();
        Stack<Integer> stack = new Stack<>();
        for(int i=0;i<prices.length;i++) stack.add(prices[i]);
        int idx = 0;
        while(!stack.empty()) {
            if(stack.peek() <= Min) {
                seconds.add(idx);
                Min = stack.peek();
            }
            else {
                if(stack.peek() > last) seconds.add(1);
                else {
                    int s = prices.length - 1 - idx;
                    int val = prices.length - 1 - s;
                    for(int i = s + 1;i < prices.length;i++) {
                        if(stack.peek() > prices[i]) {
                            val =  i - s;
                            break;
                        }
                    }
                    seconds.add(val);
                }
            }
            last = stack.pop();
            idx++;
        }
        for(int i=0;i<prices.length;i++) answer[i] = seconds.pop();
        return answer;
    }
}

File: programmers/코딩테스트 고득점 Kit/스택, 큐/주식가격.cpp
#include <string>
#include <vector>
#include <stack>
using namespace std;
vector<int> solution(vector<int> prices) {
    vector<int> answer;
    for (int i = 0; i < prices.size() - 1; i++) {
        for (int j = i + 1; j < prices.size(); j++) {
            if (prices[i] > prices[j]) {
                answer.push_back(j - i);
                break;
            }
        }
        if (answer.size() == i) answer.push_back(prices.size() - i - 1);
    }
    answer.push_back(0);
    return answer;
}